Crypt/GPG/SignatureCreationInfo.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * Part of Crypt_GPG
+ *
+ * PHP version 5
+ *
+ * @category  Encryption
+ * @package   Crypt_GPG
+ * @author    Christian Weiske <cweiske@php.net>
+ * @copyright 2015 PEAR
+ * @license   http://www.gnu.org/copyleft/lesser.html LGPL License 2.1
+ * @version   CVS: $Id$
+ * @link      http://pear.php.net/package/Crypt_GPG
+ * @link      http://pear.php.net/manual/en/package.encryption.crypt-gpg.php
+ * @link      http://www.gnupg.org/
+ */
+
+/**
+ * Information about a recently created signature.
+ *
+ * @category  Encryption
+ * @package   Crypt_GPG
+ * @author    Christian Weiske <cweiske@php.net>
+ * @copyright 2015 PEAR
+ * @license   http://www.gnu.org/copyleft/lesser.html LGPL License 2.1
+ * @link      http://pear.php.net/package/Crypt_GPG
+ * @link      http://pear.php.net/manual/en/package.encryption.crypt-gpg.php
+ * @link      http://www.gnupg.org/
+ */
+class Crypt_GPG_SignatureCreationInfo
+{
+    /**
+     * One of the three signature types:
+     * - {@link Crypt_GPG::SIGN_MODE_NORMAL}
+     * - {@link Crypt_GPG::SIGN_MODE_CLEAR}
+     * - {@link Crypt_GPG::SIGN_MODE_DETACHED}
+     *
+     * @var integer
+     */
+    protected $mode;
+
+    /**
+     * Public Key algorithm
+     *
+     * @var integer
+     */
+    protected $pkAlgorithm;
+
+    /**
+     * Algorithm to hash the data
+     *
+     * @see RFC 2440 / 9.4. Hash Algorithm
+     * @var integer
+     */
+    protected $hashAlgorithm;
+
+    /**
+     * OpenPGP signature class
+     *
+     * @var mixed
+     */
+    protected $class;
+
+    /**
+     * Unix timestamp when the signature was created
+     *
+     * @var integer
+     */
+    protected $timestamp;
+
+    /**
+     * Key fingerprint
+     *
+     * @var string
+     */
+    protected $keyFingerprint;
+
+    /**
+     * Names for the hash algorithm IDs.
+     *
+     * Names taken from RFC 3156, without the leading "pgp-".
+     *
+     * @see RFC 2440 / 9.4. Hash Algorithm
+     * @see RFC 3156 / 5. OpenPGP signed data
+     * @var array
+     */
+    protected static $hashAlgorithmNames = array(
+        1 => 'md5',
+        2 => 'sha1',
+        3 => 'ripemd160',
+        5 => 'md2',
+        6 => 'tiger192',
+        7 => 'haval-5-160',
+    );
+
+    /**
+     * Parse a SIG_CREATED line from gnupg
+     *
+     * @param string $sigCreatedLine Line beginning with "SIG_CREATED "
+     */
+    public function __construct($sigCreatedLine = null)
+    {
+        if ($sigCreatedLine === null) {
+            return;
+        }
+
+        list(
+            $title, $mode, $pkAlgorithm, $hashAlgorithm,
+            $class, $timestamp, $keyFingerprint
+        ) = explode(' ', $sigCreatedLine);
```

Comment:
Before using `list` you should check to make sure `explode` produced at least the number of required values.